### ISSUE TITLE
Create class for seeder data to be created and used in dev

### ISSUE THREAD

**Issue by robertlassila** [CodersCampus/cp#882]
This is what ChatGPT produced on how to do this:

### **1. Create a dedicated “data seeder” main class**

Put it somewhere outside your main `src/main/java` entry point package to avoid confusion with the normal main class.

```java
package com.example.dataseeder;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication(scanBasePackages = "com.example") // scan your app's packages
public class TestDataSeederApplication {

    public static void main(String[] args) {
        SpringApplication.run(TestDataSeederApplication.class, args);
    }
}
```

---

### **2. Add a runner that creates the test data**

Use `CommandLineRunner` or `ApplicationRunner` so the logic runs after the context is up, and then exit the app.

```java
package com.example.dataseeder;

import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class TestDataLoader implements CommandLineRunner {

    private final UserRepository userRepository;
    private final MusicDocumentRepository musicDocumentRepository;

    public TestDataLoader(UserRepository userRepository,
                          MusicDocumentRepository musicDocumentRepository) {
        this.userRepository = userRepository;
        this.musicDocumentRepository = musicDocumentRepository;
    }

    @Override
    public void run(String... args) throws Exception {
        // Example: Clear existing test data
        musicDocumentRepository.deleteAll();
        userRepository.deleteAll();

        // Insert test data
        User user = new User();
        user.setName("Test User");
        user.setEmail("test@example.com");
        userRepository.save(user);

        MusicDocument doc = new MusicDocument();
        doc.setTitle("Sample Score");
        doc.setUser(user);
        musicDocumentRepository.save(doc);

        System.out.println("✅ Test data created");

        // Exit after seeding
        System.exit(0);
    }
}
```

---

### **3. Run it separately**

* In your IDE, **right-click** on `TestDataSeederApplication` → **Run**.
* It will spin up Spring, insert your data, and exit.
* Your main application stays untouched.

---

### **4. (Optional) Keep it out of production**

You can make sure this seeder **only runs** when you want by using a Spring profile:

```java
@Component
@Profile("dataseed")
public class TestDataLoader implements CommandLineRunner { ... }
```

Then run it with:

```bash
mvn spring-boot:run -Dspring-boot.run.profiles=dataseed -Dspring-boot.run.main-class=com.example.dataseeder.TestDataSeederApplication
```

---

**Summary:**
You’ll have a second main class (`TestDataSeederApplication`) + a `CommandLineRunner` bean that inserts your test data and then calls `System.exit(0)`. This way, you can run **only the seeding logic** without booting your main app’s web server.

---

**Comment by robertlassila** [comment]
This is what ChatGPT gave about running a specific profile on your IDE

1. Using application-<profile>.properties or application-<profile>.yml

Spring Boot automatically picks up application.properties or application.yml and merges in any application-<profile>.properties (or .yml) if you activate that profile.

2. Activating a profile in IntelliJ IDEA

There are two main ways:

A. Through Run/Debug Configurations

Go to Run → Edit Configurations…

Select your Spring Boot application configuration.

In the VM options field, add:

-Dspring.profiles.active=prod


Replace prod with the profile you want (dev, test, etc.).

Apply and Run.

This tells Spring Boot to only use the application-prod.properties (or application-prod.yml) in addition to application.properties.

B. Through Program Arguments

Alternatively, in the same Run/Debug Configurations window:

In Program arguments, add:

--spring.profiles.active=prod


Apply and Run.

This is equivalent to the VM option method but works at the application argument level.

3. Activating a profile in Eclipse

Right-click your Spring Boot class → Run As → Run Configurations…

Select your application → Arguments tab.

Under VM arguments, add:

-Dspring.profiles.active=prod


Or under Program arguments, add:

--spring.profiles.active=prod


Apply and Run.

4. Quick sanity check

Inside your code, you can log the active profiles to verify:

@SpringBootApplication
public class DemoApplication {

    public static void main(String[] args) {
        ConfigurableApplicationContext ctx = SpringApplication.run(DemoApplication.class, args);
        System.out.println("Active profiles: " + Arrays.toString(ctx.getEnvironment().getActiveProfiles()));
    }
}


If you want, I can give a shortcut way to quickly switch profiles in IntelliJ without editing configurations every time, which is handy for dev/test/prod switching.

Do you want me to show that?

**Comment by robertlassila** [comment]
- Make a new controller and add a get mapping for triggering the creation of the data with a  url
- Use application properties as a fail safe to keep the trigger from working in prod
- Make it idempotent, meaning don't create multiple copies of the data on one database
- Add another get mapping in the controller we made that will delete the seeder data (its own service)

**Comment by sayazian** [comment]
In order to inject the seed class data we can follow the below approach:

In the application.properties:

seed.data=true


In your service class:

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class MyService {

    @Value("${seed.data}")
    private boolean seedData;

    public void runSeedData() {
        if (seedData) {
            System.out.println("Data is injected!");
        } else {
            System.out.println("Data is not injected.");
        }
    }
}
